sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -1483,8 +1508,14 @@ def create_pipeline_spec(
     pipeline_spec.root.CopyFrom(
         _build_component_spec_from_component_spec_structure(component_spec))
 
+    # TODO: add validation of returned outputs -- it's possible to return
+    # an output from a task in a condition group, for example, which isn't
+    # caught until submission time using Vertex SDK client
```

Comment:
Note: This issue existed prior to this PR. The following pipeline can currently be compiled, but may not return the output `int` it promises.

```
@dsl.pipeline
def p(x: int) -> int:
    with dsl.Condition(x == 1):
        return c().output
```